htmresearch/regions/LanguageSensor.py

```diff
@@ -162,37 +162,33 @@ def compute(self, inputs, outputs):
     outputs are as defined in the spec above.
 
     Expects the text data to be in under header "token" from the dataSource.
-
-    TODO: validate we're handling resets correctly
     """
     if len(self.queue) > 0:
-      # data has been added to the queue, so use it
+      # Data has been added to the queue, so use it
       data = self.queue.pop()
 
     elif self.dataSource is None:
       raise Exception("LanguageSensor: No data to encode: queue is empty "
                         "and the dataSource is None.")
     else:
       data = self.dataSource.getNextRecordDict()
-      # Keys in data that are not column headers from the data source are standard
-      # of RecordStreamIface objects.
+      # Keys in data that are not column headers from the data source are
+      # standard of RecordStreamIface objects
 
-    # Copy important data input fields over to outputs dict. We set "sourceOut"
-    # explicitly b/c PyRegion.getSpec() won't take an output field w/ type str.
+    # Copy important data input fields over to outputs dict.
     outputs["resetOut"][0] = data["_reset"]
     outputs["sequenceIdOut"][0] = data["_sequenceId"]
-    outputs["sourceOut"] = data["_token"]
-    self.populateCategoriesOut(data["_category"], outputs['categoryOut'])
-    outputs["encodingOut"] = self.encoder.encodeIntoArray(
-      data["_token"], outputs["dataOut"])
+    self.populateCategoriesOut(data["_category"], outputs["categoryOut"])
+    self.encoder.encodeIntoArray(data["_token"], outputs["dataOut"])
 
     if self.verbosity > 0:
       print "LanguageSensor outputs:"
       print "SeqID: ", outputs["sequenceIdOut"]
       print "Categories out: ", outputs['categoryOut']
-      print "dataOut: ",outputs["dataOut"].nonzero()[0]
+      print "dataOut: ", outputs["dataOut"].nonzero()[0]
 
-    self._outputValues = copy.deepcopy(outputs)
+    self._outputValues = {field: value for field, value in outputs.iteritems()}
```

Comment:
OK, that makes sense. Then my guess is you need to do {field: copy.deepcopy(value) ...}  The arrays in the outputs dict correspond to fixed memory buffers in C++ that are setup by the Network API, so if you want to store them you have to do a copy.